### PR TITLE
Add variable annotations to `optuna/*.py`

### PR DESCRIPTION
**Pull request by akihironitta**
## Motivation
See #1897.

## Description of the changes
Adds variable annoations to:
- optuna/_imports.py
- optuna/dashboard.py
- optuna/progress_bar.py
- optuna/study.py

### CODE REVIEW

**Comment by akihironitta** [comment]
@HideakiImamura Do we need type annotations for temporary local variables as well?

For example, `need_to_add_callback` in this case:
https://github.com/optuna/optuna/blob/5940fd75059c469a729ec66ee84925f8c2e12ed7/optuna/dashboard.py#L230-L232

**Comment by crcrpar** [comment]
`need_to_add_callback` don't need to get annotated as it's const-ish variable and its type is explicit enough to me 😉

File: optuna/dashboard.py [hunk 1]
```diff
@@ -57,7 +58,7 @@
 </p>
 """
 
-_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+_DATETIME_FORMAT: str = "%Y-%m-%d %H:%M:%S"
```
Comment (hvy) [inline optuna/dashboard.py:61]: I'm wondering if we should avoid introducing type hints to variables that didn't used to have comment style type hints. We usually only add type hints where mypy would complain assuming that it's clever to detect possible bug-prone code (it get a bit verbose and may hurt readability too adding type hints). It also seems a bit orthogonal to the motivation behind #1897 but sorry, this was probably not communicated well enough. What do you think?

Comment (akihironitta) [reply optuna/dashboard.py:61]: (Although I have added a lot of annotations already,) I have to admit that I strongly agree with you that too many annotations lower readability, and it sounds reasonable to add variable annotations only to variables which already have comment style annotations. I'll roll back some commits if you're ok with it!

Comment (akihironitta) [reply optuna/dashboard.py:61]: If we add anntations only to variables which already have comment style annotations, we can just use `com2ann` suggested by @harupy in the issue all at once!

Comment (hvy) [reply optuna/dashboard.py:61]: Sure, that might be better using `com2ann`. Do you want to close this PR and create a new one?

Comment (akihironitta) [reply optuna/dashboard.py:61]: Sure.

Comment (hvy) [reply optuna/dashboard.py:61]: Looking forward, thanks!

**Pull request closed without merging**